protocol/communities/manager.go

```diff
@@ -256,11 +256,11 @@ func NewManager(identity *ecdsa.PrivateKey, installationID string, db *sql.DB, e
 		torrentConfig:               torrentConfig,
 		torrentTasks:                make(map[string]metainfo.Hash),
 		historyArchiveDownloadTasks: make(map[string]*HistoryArchiveDownloadTask),
-		persistence: &Persistence{
-			logger:     logger,
-			db:         db,
-			timesource: timesource,
-		},
+	}
+
+	manager.persistence = &Persistence{
+		db:                      db,
+		recordBundleToCommunity: manager.dbRecordBundleToCommunity,
```

Comment:
it's too late, but this recordBundleToCommunity callback is not a good idea. What gonna happen if someone decides to override this callback? It can lead to unexpected behavior as it will be used on every all other queries

Reply:
Not sure I am following. It is encapsulated into the structure like everything else. The same question I can ask about the `db`, what gonna happen if someone decides to override `db` field with something else?

Can you please elaborate more of what you meant, what would be the motivation to override this callback?